### ISSUE TITLE
Reddit oAuth url is for desktop and doesnt redirect back if user needs to sign in

### ISSUE THREAD

**Issue by Adeering** [draw-dev/DRAW#182]
Found an issue where on mobile devices (haven't tried on computers or web) that if the user isnt already logged in to Reddit after they log in the authUrl doesnt redirect them back to the authorization page.  I found that this is an issue with the reddit side of things but found a workaround that might be useful to integrate.  Either a mobile bool flag in the reddit.auth.url function in draw or just add it in.  Here is my fix:

```
String authUrl = reddit.auth.url(["read", "edit", "identity", "privatemessages", "submit", ], "scryer-auth").toString();
authUrl = authUrl.replaceFirst('authorize', 'authorize.compact');
```
Basically just need to add '.compact' after authorize in the url

`https://www.reddit.com/api/v1/authorize.compact?resp....`

**Comment by StefanLobbenmeier** [comment]
Is the compact parameter what you are looking for?


```dart
    final authUrl = _reddit.auth.url(
      ['*'],
      state,
      compactLogin: true,
    );
```

**Comment by bkonyi** [comment]
@StefanLobbenmeier's response is the right answer here.
